content/docs/develop/userscripts/about-userscripts.md

```diff
@@ -0,0 +1,142 @@
+
+Even though Scratch Addons userscripts are part of a browser extension, they cannot access any `chrome.*` or `browser.*` APIs. Instead, Scratch Addons offers an [`addon.*` API](/docs/reference/addon-api/). 
+
+
+## Declaring userscripts in the addon manifest
+
+{{< admonition warning >}}
+**Some changes require an extension reload** from `chrome://extensions` to take effect, such as updating the addon manifest file.
+
+It's not necessary to reload the extension when changing the source of an already existing userscript JavaScript file. In those cases, reloading the page is enough.
+{{< /admonition >}}
+
+<!-- TODO: revamp -->
+
+<!-- TODO: create a "matches" page -->
+
+Go to the manifest of your addon (addon.json) and add a property called `userscripts"`.  
+This property must be an array.  
+Each item of the array must have the following properties: `"url"` and `"matches"`.  
+`"url"` must be a relative URL to a JavaScript file.  
+`"matches"` must be an array of URLs where you want to run the userscript on. You can use asterisks.
+Example manifest:
+```json
+{
+  "name": "Scratch Messaging",
+  "description": "Provides easy reading and replying to your Scratch messages.",
+  "userscripts": [
+    {
+      "url": "userscript.js",
+      "matches": ["https://scratch.mit.edu/*"]
+    },
+    {
+      "url": "second_userscript.js",
+      "matches": ["https://scratch.mit.edu/projects/*", "https://scratch.mit.edu/users/*"]
+    }
+  ],
+  "tags": ["community"],
+  "enabledByDefault": false
+}
+```
+
+## Creating your first userscript
+
+Unlike extension content scripts and Tampermonkey userscripts, you must wrap all of your code inside a module default export:
+```js
+// Example userscript
+export default async function ({ addon, console }) {
+  console.log("Hello, " + await addon.auth.fetchUsername());
+  console.log("How are you today?");
+}
+```
+
+Remember that JavaScript allows functions to be declared inside other functions, for example:
+```js
+export default async function ({ addon, console }) {
+  async function sayHelloToUser() {
+    console.log("Hello, " + await addon.auth.fetchUsername());
+  }
+
+  await sayHelloToUser();
+  console.log("How are you today?");
+}
+```
+
+{{< admonition info >}}
+You can access many `addon.*` API utilities from userscripts. For example, you can get the current username, wait until an element exists on the page, or get a reference to the Scratch VM object.
+
+For more information, check the [API reference](/docs/reference/addon-api/).
+{{< /admonition >}}
+
+
+## Modifying the document HTML
+
+Use [browser DOM APIs](https://developer.mozilla.org/en-US/docs/Web/API/HTML_DOM_API) to customize the HTML of the page.
+
+<!-- TODO : an example -->
+
+## Localizing userscripts
+
+Addon userscripts sometimes need to reference English words or sentences. Make sure not to hardcode them in the code, so that they can be part of the translation process.
```

Comment:
```suggestion
Addon userscripts sometimes need to reference English words or sentences. Make sure not to hardcode them, so that they can be part of the translation process.
```